src/bson.c

```diff
@@ -717,6 +717,11 @@ void object_to_bson(zval *object, php_phongo_bson_flags_t flags, const char *key
 			return;
 		}
 	}
+
+	/* Even if we don't know how to encode the object, ensure that we at least
+	 * create an empty BSON document. */
+	bson_append_document_begin(bson, key, key_len, &child);
+	bson_append_document_end(bson, &child);
```

Comment:
https://github.com/mongodb-labs/mongo-php-driver-prototype/pull/103/files#diff-f7d29a680148f52d6601f59ed787f577R712 throws an exception for an unsupported BSON Type, but any other object will ideally get its public properties serialized.